src/adapter/src/coord/sequencer.rs

```diff
@@ -1894,11 +1894,39 @@ impl<S: Append + 'static> Coordinator<S> {
 
         let vars = session.vars_mut();
         let (name, local) = (plan.name, plan.local);
-        match plan.value {
-            SetVariableValue::Default => vars.reset(&name, local)?,
-            SetVariableValue::Literal(Value::String(s)) => vars.set(&name, &s, local)?, // pass-through unquoted strings
-            SetVariableValue::Ident(ident) => vars.set(&name, &ident.into_string(), local)?, // pass-through unquoted idents
-            value => vars.set(&name, &value.to_string(), local)?, // or else use the AstDisplay for SetVariableValue
+        let value = match plan.value {
+            SetVariableValue::Default => None,
+            SetVariableValue::Literal(Value::String(s)) => Some(s), // unquoted strings
+            SetVariableValue::Ident(ident) => Some(ident.into_string()), // pass-through unquoted idents
+            value => Some(value.to_string()), // or else use the AstDisplay for SetVariableValue
+        };
+
+        match value.clone() {
+            Some(v) => vars.set(&name, &v, local)?,
+            None => vars.reset(&name, local)?,
+        }
+
+        // Add notice if database or cluster value does not correspond to a catalog item.
+        if let Some(v) = value {
+            match name.as_str() {
+                "database" => {
+                    if matches!(
+                        self.catalog.resolve_database(&v),
+                        Err(CatalogError::UnknownDatabase(_))
+                    ) {
+                        session.add_notice(AdapterNotice::DatabaseDoesNotExist { name: v });
+                    }
+                }
+                "cluster" => {
+                    if matches!(
+                        self.catalog.resolve_compute_instance(&v),
+                        Err(CatalogError::UnknownComputeInstance(_))
+                    ) {
+                        session.add_notice(AdapterNotice::ClusterDoesNotExist { name: v });
+                    }
+                }
+                &_ => {}
```

Comment:
nit: ref to a wildcard is weird!

```suggestion
                _ => {}
```

Reply:
Thanks! For some reason the compiler suggested that to me at some point so I assumed it was a best practice.